addons/account/static/tests/legacy/x2many_buttons_tests.js

```diff
@@ -96,22 +96,23 @@ QUnit.module("Fields", (hooks) => {
     });
 
     QUnit.test("edit record and check if edits get discarded when click on one of the buttons and redirects to proper record", async function (assert) {
-        assert.expect(3);
-        const form = await makeView({
+        assert.expect(2);
+        await makeView({
             resModel: "account.move",
             resId: 1,
             serverData,
             arch: fromArch,
             type: "form",
-        });
-        patchWithCleanup(form.env.services.action, {
-            doAction(action) {
-                assert.deepEqual(action, {
-                    res_model: "account.move",
-                    res_id: 2,
-                    type: "ir.actions.act_window",
-                    views: [[false, "form"]],
-                });
+            mockRPC(route) {
+                if (route === "/web/dataset/call_kw/account.move/action_open_business_doc") {
+                    return {
+                        name: "action",
+                        res_model: "account.move",
+                        res_id: 2,
+                        type: "ir.actions.act_window",
+                        views: [[false, "form"]],
+                    };
+                }
```

Comment:
Before the action was hard coded in JS but since I changed that following wan's comment, now we do an ORM call, so we need to mock the action. 

The assertions that come after the mockRPC are the same though, but I did add some asserts to the mockRPC call as well. 

But maybe I didn't understand your comment fully, so I'll double check with you!